lib/gelf/notifier.rb

```diff
@@ -7,26 +7,32 @@ class << self
     end
 
     attr_accessor :enabled, :collect_file_and_line, :rescue_network_errors
-    attr_reader :max_chunk_size, :level, :default_options, :level_mapping
+    attr_reader :max_chunk_size, :level, :default_options, :config_options, :level_mapping
 
     # +host+ and +port+ are host/ip and port of graylog2-server.
     # +max_size+ is passed to max_chunk_size=.
     # +default_options+ is used in notify!
-    def initialize(host = 'localhost', port = 12201, max_size = 'WAN', default_options = {})
+    def initialize(host = 'localhost', port = 12201, max_size = 'WAN', default_options = {}, config_options = {})
       @enabled = true
       @collect_file_and_line = true
 
+      @@options = config_options
```

Comment:
Why are you using a class variable here? This will break if you create multiple instances of the notifier/logger. I think we can just pass the `config_options` to the `RubyTcpSender` (notifier.rb, line 37) and `RubyUdpSender` (notifier.rb, line 39) to avoid using a class variable.